File: sdks/python/blacklake/exceptions.py
"""
BlackLake Python SDK Exceptions

Custom exceptions for the BlackLake SDK.
"""


class BlackLakeError(Exception):
    """Base exception for BlackLake SDK errors."""
    
    def __init__(self, message: str, details: dict = None):
        super().__init__(message)
        self.message = message
        self.details = details or {}


class AuthenticationError(BlackLakeError):
    """Authentication failed."""
    pass


class AuthorizationError(BlackLakeError):
    """Access denied."""
    pass


class NotFoundError(BlackLakeError):
    """Resource not found."""
    pass


class ValidationError(BlackLakeError):
    """Request validation failed."""
    pass


class RateLimitError(BlackLakeError):
    """Rate limit exceeded."""
    pass


class ServerError(BlackLakeError):
    """Server error."""
    pass


class NetworkError(BlackLakeError):
    """Network error."""
    pass


class TimeoutError(BlackLakeError):
    """Request timeout."""
    pass

File: sdks/typescript/src/exceptions.ts
/**
 * BlackLake TypeScript SDK Exceptions
 * 
 * Custom exceptions for the BlackLake SDK.
 */

export class BlackLakeError extends Error {
  public readonly details?: Record<string, any>

  constructor(message: string, details?: Record<string, any>) {
    super(message)
    this.name = 'BlackLakeError'
    this.details = details
  }
}

export class AuthenticationError extends BlackLakeError {
  constructor(message: string = 'Authentication failed', details?: Record<string, any>) {
    super(message, details)
    this.name = 'AuthenticationError'
  }
}

export class AuthorizationError extends BlackLakeError {
  constructor(message: string = 'Access denied', details?: Record<string, any>) {
    super(message, details)
    this.name = 'AuthorizationError'
  }
}

export class NotFoundError extends BlackLakeError {
  constructor(message: string = 'Resource not found', details?: Record<string, any>) {
    super(message, details)
    this.name = 'NotFoundError'
  }
}

export class ValidationError extends BlackLakeError {
  constructor(message: string = 'Validation failed', details?: Record<string, any>) {
    super(message, details)
    this.name = 'ValidationError'
  }
}

export class RateLimitError extends BlackLakeError {
  constructor(message: string = 'Rate limit exceeded', details?: Record<string, any>) {
    super(message, details)
    this.name = 'RateLimitError'
  }
}

export class ServerError extends BlackLakeError {
  constructor(message: string = 'Server error', details?: Record<string, any>) {
    super(message, details)
    this.name = 'ServerError'
  }
}

export class NetworkError extends BlackLakeError {
  constructor(message: string = 'Network error', details?: Record<string, any>) {
    super(message, details)
    this.name = 'NetworkError'
  }
}

export class TimeoutError extends BlackLakeError {
  constructor(message: string = 'Request timeout', details?: Record<string, any>) {
    super(message, details)
    this.name = 'TimeoutError'
  }
}
